File: app/src/test/java/dev/bee/kanjianki/RunnableClickListenerTest.java
package dev.bee.kanjianki;

import org.junit.Test;

import java.util.concurrent.atomic.AtomicBoolean;

import static org.junit.Assert.assertTrue;

public final class RunnableClickListenerTest {
    @Test
    public void onClickRunsActionWhenViewIsNull() {
        AtomicBoolean clicked = new AtomicBoolean(false);

        new RunnableClickListener(() -> clicked.set(true)).onClick(null);

        assertTrue(clicked.get());
    }

    @Test(expected = NullPointerException.class)
    public void nullActionFailsWhenClicked() {
        new RunnableClickListener(null).onClick(null);
    }
}

File: app/src/test/java/dev/bee/kanjianki/RunnableClickListenerTest.kt
package dev.bee.kanjianki

import org.junit.Assert.assertTrue
import org.junit.Test
import java.util.concurrent.atomic.AtomicBoolean

class RunnableClickListenerTest {
    @Test
    fun onClickRunsActionWhenViewIsNull() {
        val clicked = AtomicBoolean(false)

        RunnableClickListener(Runnable { clicked.set(true) }).onClick(null)

        assertTrue(clicked.get())
    }

    @Test(expected = NullPointerException::class)
    fun nullActionFailsWhenClicked() {
        RunnableClickListener(null).onClick(null)
    }
}
